src/axom/inlet/Table.cpp

```diff
@@ -487,47 +499,57 @@ struct PrimitiveArrayHelper<Key, bool>
    * \param [inout] table The table to add the collection to
    * \param [in] reader The Reader object to read the collection from
    * \param [in] lookupPath The path within the input file to the collection
+   * 
+   * \return The keys from the collection that was added
    *****************************************************************************
    */
-  PrimitiveArrayHelper(Table& table, Reader& reader, const std::string& lookupPath)
+  static std::vector<VariantKey> add(Table& table,
```

Comment:
This is probably out of scope for this PR, but does this need to be a class? This seems like a strange design. Can't this just be accomplished with function templates? This applies to to all the `PrimitiveArrayHelper` templates.

Reply:
I agree that this looks a little suspect, but this ended up being necessary as I needed partial template specializations to minimize duplication, which C++ doesn't allow for function templates (only class templates).